x-pack/plugins/fleet/server/types/models/preconfiguration.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import semverValid from 'semver/functions/valid';
+
+const varsSchema = schema.maybe(
+  schema.arrayOf(
+    schema.object({
+      name: schema.string(),
+      type: schema.maybe(schema.string()),
+      value: schema.oneOf([schema.string(), schema.number()]),
+    })
+  )
+);
+
+export const PreconfiguredPackagesSchema = schema.arrayOf(
+  schema.object({
+    name: schema.string(),
+    version: schema.string({
+      validate: (value) => {
+        if (!semverValid(value)) {
+          return i18n.translate('xpack.fleet.config.invalidPackageVersionError', {
+            defaultMessage: 'must be a valid semver',
+          });
+        }
+      },
+    }),
+  })
+);
+
+export const PreconfiguredAgentPoliciesSchema = schema.arrayOf(
```

Comment:
I wonder if it is possible to de-dupe some of these schema by pulling in existing definitions from `models/package_policy.ts` and `models/agent_policy.ts`?

for example, `namespace` could be `schema.maybe(AgentPolicyBaseSchema.namespace)`
(namespace is a good example because we have more validation on that besides just enforcing string type)